library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt

```diff
@@ -4,13 +4,18 @@ package io.envoyproxy.envoymobile
  * Specifies how a request may be retried, containing one or more rules.
  *
  * @param maxRetryCount Maximum number of retries that a request may be performed.
- * @param retryOn Whitelist of rules used for retrying.
+ * @param retryOn Whitelist of rules used for retrying. Must be <= `totalUpstreamTimeoutMS`
```

Comment:
is it worth enforcing this somehow? Or at least warning? I don't know what is convention in client land

Reply:
We could, though since this validation is done within Envoy itself, it seems like maybe that should be done there?

Reply:
Envoy will just silently ignore. I guess the case that is the least intuitive and could cause people headache is if they are unaware that the default is 15s and they set a per try of >15s and they see timeouts happening at 15s, but had no warning (or appropriate mobile convention).

Reply:
Talked to @buildbreaker and we decided to throw here. Updated